moveit_commander/src/moveit_commander/planning_scene_interface.py

```diff
@@ -306,14 +239,59 @@ def __make_cylinder(name, pose, height, radius):
         return co
 
     @staticmethod
-    def __make_planning_scene_diff_req(collision_object, attach=False):
+    def __make_plane(self, name, pose, normal=(0, 0, 1), offset=0):
+        co = CollisionObject()
+        co.operation = CollisionObject.ADD
+        co.id = name
+        co.header = pose.header
+        p = Plane()
+        p.coef = list(normal)
+        p.coef.append(offset)
+        co.planes = [p]
+        co.plane_poses = [pose.pose]
+        return co
+
+    @staticmethod
+    def __make_planning_scene_diff_req(collision_object):
         scene = PlanningScene()
         scene.is_diff = True
         scene.robot_state.is_diff = True
-        if attach:
+        if isinstance(collision_object, AttachedCollisionObject):
             scene.robot_state.attached_collision_objects = [collision_object]
         else:
             scene.world.collision_objects = [collision_object]
         planning_scene_diff_req = ApplyPlanningSceneRequest()
         planning_scene_diff_req.scene = scene
         return planning_scene_diff_req
+
+
+def __create_generic_methods(shape):
+    """Add methods add_xxx() and attach_xxx() to class PlanningSceneInterface for a given shape"""
+    make_method = getattr(PlanningSceneInterface, '_PlanningSceneInterface__make_' + shape)
+
+    def add(self, name, pose, *args, **kwargs):
+        co = make_method(name, pose, *args, **kwargs)
+        self._PlanningSceneInterface__submit(co)
+
+    def attach(self, link, name, pose=None, *args, **kwargs):
+        touch_links = kwargs.pop('touch_links', [link])
```

Comment:
Doesn't this [raise a KeyError](https://stackoverflow.com/questions/49218302/python-difference-between-kwargs-pop-and-kwargs-get) if `touch_links` is given as an unnamed argument?

Reply:
No, because I provide a default value.

Reply:
So I guess there is no way to specify touch_links as an **unnamed** argument (as Felix asked)?

Reply:
Of course, there is (as you probably just figured out): If the argument is provided as last, unnamed argument, the first call will fail with a TypeError and we try again.

Reply:
Yes I did, sorry for the noise.

Relying on the fact that the `__make` function family does not accept a list of string argument seems rather implicit though, and I guess that' what confused @felixvd as well.

Could you please add a comment that explains *why* the exception is thrown and not just *that* it is thrown?